src/oic/utils/keyio.py

```diff
@@ -777,6 +789,39 @@ def get_issuer_keys(self, issuer):
             res.extend(kbl.keys())
         return res
 
+    def __eq__(self, other):
+        if not isinstance(other, KeyJar):
+            return False
+
+        # The set of issuers MUST be the same
+        if set(self.keys()) != set(other.keys()):
+            return False
+
+        # Keys per issuer must be the same
+        for iss in self.keys():
+            sk = self.get_issuer_keys(iss)
+            ok = other.get_issuer_keys(iss)
+            if len(sk) != len(ok):
```

Comment:
The set of keys must be that same which means that the number of keys must be the same.
That's a MUST but not sufficient as the lines below show.
It's an optimisation. If the number of keys are different then I don't have to check the
individual keys.